crates/ruff_python_formatter/src/expression/expr_attribute.rs

```diff
@@ -27,25 +27,28 @@ impl FormatNodeRule<ExprAttribute> for FormatExprAttribute {
             })
         );
 
+        let comments = f.context().comments().clone();
+        let dangling_comments = comments.dangling_comments(item);
+        let leading_attribute_comments_start =
+            dangling_comments.partition_point(|comment| comment.line_position().is_end_of_line());
+        let (trailing_dot_comments, leading_attribute_comments) =
+            dangling_comments.split_at(leading_attribute_comments_start);
+
         if needs_parentheses {
             value.format().with_options(Parenthesize::Always).fmt(f)?;
+        } else if let Expr::Attribute(expr_attribute) = value.as_ref() {
+            // We're in a attribute chain (`a.b.c`). The outermost node adds parentheses if
+            // required, the inner ones don't need them so we skip the `Expr` formatting that
+            // normally adds the parentheses.
+            expr_attribute.format().fmt(f)?;
         } else {
```

Comment:
Can you add a test case where an inner attribute change is parenthesized? I wonder if we need to preserve the parentheses.